independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/options/BootstrapMavenOptions.java

```diff
@@ -110,7 +115,74 @@ public static BootstrapMavenOptions newInstance(String cmdLine) {
 
     public static String getMavenCmdLine() {
         final String mvnCmd = PropertyUtils.getProperty(QUARKUS_INTERNAL_MAVEN_CMD_LINE_ARGS);
-        return mvnCmd == null ? System.getenv(MAVEN_CMD_LINE_ARGS) : mvnCmd;
+        return mvnCmd == null || mvnCmd.isEmpty() ? System.getenv(MAVEN_CMD_LINE_ARGS) : mvnCmd;
```

Comment:
I think if the property is not set and we are returning the envvar, which we know isn't correct, we could do the quoting work in this method. Instead of adding these updateMavenCmdLineXXX methods

Reply:
Maybe but using a separate method affords us the possibility of returning the quoted user-properties in the context where it's needed, instead of having to redo essentially the same work at a later point.

Reply:
Done.